Enable systemd services-nilrt

usercw88 opened the pull request:
### Summary of Changes

Create systemd equivalent of initscripts-nilrt-safemode -- services-nilrt-safemode
Create systemd equivalent of initscripts-nilrt -- services-nilrt

### Justification

[AB#2572995](https://dev.azure.com/ni/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/2572995)
[AB#2572980](https://dev.azure.com/ni/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/2572980)

### Testing

* [x] I have built the core package feed with this PR in place. (`bitbake packagefeed-ni-core`)
* [x] I have built the index package feed with this PR in place. (`bitbake package-index`)
* [x] I have built the safemode images with this PR in place. (`bitbake nilrt-safemode-rootfs`)
* [x] Verified services-nilrt-safemode services are enabled and started.
* [x] I have built the base system image with this PR in place. (`bitbake nilrt-base-system-image`)
* [x] I have applied the systemd nilrt-base-system-image on top of a sysv nilrt-safemode-rootfs
  * [x] Verified servies-nilrt services are enabled and started.


### Procedure

* [x] I certify that the contents of this pull request complies with the [Developer Certificate of Origin](https://github.com/ni/nilrt/blob/HEAD/docs/CONTRIBUTING.md#developer-certificate-of-origin-dco).

usercw88 committed:
services-nilrt-safemode: Created recipe framework

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
mountcompatibility: Add to services-nilrt-safemode

mountcompatibility: Identical to mountcompaibility init.d script in
                    recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
mountuserfs: Add to services-nilrt-safemode

mountuserfs: Identical to mountuserfs init.d script from
             recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
nisafemodereason: Add to services-nilrt-safemode

nisafemodereason: Identical to nisafemodereason init.d script from
                  recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
niselectnetnaming: Add to services-nilrt-safemode

niselectnetnaming: Identical to niselectnetnaming init.d script from
                   recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
services-nilrt: Add recipe framework

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
cleanvarcache: Add to service-nilrt

cleanvarcache.service: ExecStart mimics cleanvarcache init.d script from
                       recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
firewall: Add firewall service to service-nilrt

firewall: Identical to firewall init.d script from
          recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
iso3166-translation: Install via services-nilrt

iso3166-translation.txt: Identical to iso3166-translation.txt from
                         recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
lvrt-cgroup: Add scripts to services-nilrt

lvrt-cgroup: Identical to lvrt-cgroup file from recipes-ni/initscripts-nilrt
lvrt-cgroup.sh: Identical to lvrt-cgroup.sh from
                recipes-ni/initscriptsnilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
mountconfig: Add service to services-nilrt

mountconfig: Identical to mountconfig init.d script from
             recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
nicheckbiosconfig: Add service to services-nilrt

nicheckbiosconfig: Identical to nicheckbiosconfig init.d script from
                   recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
nicleanefivars: Add service to services-nilrt

nicleanefivars.service: ExecStart identical to nicleanefivars init.d
			script from recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
nicleanstalelinks: Add service to services-nilrt

nicleanstalelinks.service: ExecStart identical to init.d nicleanstalelinks
                           from recipes-ni/initscript-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
nicreatecpuacctgroups: Add to services-nilrt

nicreatecpuacctgroups: Modified from nicreatecpuacctgroups init.d script
                       from recipes-ni/initscripts-nilrt
  - Path to lvrt-cgroup.sh updated to /usr/share/services-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
nicreatecpusets: Add service to services-nilrt

nicreatecpusets: Modified from recipes-ni/initscripts-nilrt
  - Update path to lvrt-cgroups to /usr/share/services-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
nidisablecstates: Add service to services-nilrt

nidisablecstates.service: ExecStart identical to init.d script from
                          recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
nisetcommitratio: Add service to services-nilrt

nisetcommitratio: Identical to nisetcommitratio init.d script from
                  recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
nisetreboottype: Add service to services-nilrt

nisetreboottype: Identical to nisetreboottype init.d script from
                 recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
nisetupkernelconfig: Add service to services-nilrt

nisetupkernelconfig: Identical to nisetupkernelconfig init.d script
                     from recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
nipopulateconfigdir: Add service to services-nilrt

nipopulateconfigdir.service: ExecStart identical to init.d script
                             from recipes-ni/initscript-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
populateconfig: Add service to services-nilrt

populateconfig: Identical to populateconfig init.d script from
                recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
wirelesssetdomain: Add service to services-nilrt

wirelesssetdomain: Identical to wirelesssetdomain init.d script
                   from recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
packagegroups: Install services-nilrt

ni-netcfgutil: Install services-nilrt on systemd systems

Signed-off-by: Can Wong <can.wong@emerson.com>

usercw88 committed:
systctl.conf: Disable fs.protected_regular

nirtcfg fails to initialize when fs.protected_regular is enabled - see AB#3053779

Signed-off-by: Can Wong <can.wong@emerson.com>

chaitu236 commented on recipes-ni/services-nilrt/files/niselectnetnaming.service:
```diff
@@ -0,0 +1,12 @@
+ExecStart=/lib/systemd/scripts/niselectnetnaming
+
+[Install]
+WantedBy=multi-user.target
```
Missing newline at end

usercw88 replied:
newline added

chaitu236 reviewed (commented):
Haven't looked at this PR closely yet but before I do, wanted to know if all the non-service files in `services-nilt/files` are unmodified copies from initscripts-nilrt recipe.

usercw88 commented:
> Haven't looked at this PR closely yet but before I do, wanted to know if all the non-service files in `services-nilt/files` are unmodified copies from initscripts-nilrt recipe.

Yes, the scripts are the same as the ones in `initscripts-nilrt` with the exception of `nicreatecpuacctgroups` and `nicreatecpusets`. The changes to those files were to update the references to the `lvrt-group.sh` script, which is installed to a package specific location.

chaitu236 commented on recipes-ni/services-nilrt/files/mountdebugfs:
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
```
We changed the implementation for this and renamed it in https://github.com/ni/meta-nilrt/pull/810.

usercw88 replied:
Reverted the change that added mountdebugfs and added niconfigtracefs service

chaitu236 commented on recipes-ni/services-nilrt/files/nipopulateconfigdir.service:
```diff
@@ -0,0 +1,16 @@
+[Service]
+Type=oneshot
+ExecStart=/bin/bash -c '\
+  for i in /etc/populateconfig.d/*.sh; do \
```
Having script in service file looks odd. Is this how it's done in existing systemd recipes?

chaitu236 replied:
Even if this isn't the standard, we don't have to fix this now as we're just trying to standup systemd. We could have a TD item/task to address later.

usercw88 replied:
I think this will be flushed out when we "Create "services" standards document", see [AB#3067555](https://dev.azure.com/ni/DevCentral/_workitems/edit/3067555)

chaitu236 reviewed (commented):
Looks ok but there is an open item and a question from last review. Do you plan to address them in another PR?

usercw88 committed:
niconfiguretracefs: Add to services-nilrt

niconfiguretracefs: Identical to niconfiguretracefs init.d script in
                    recipes-ni/initscripts-nilrt

Signed-off-by: Can Wong <can.wong@emerson.com>

chaitu236 commented on recipes-ni/services-nilrt/files/niconfiguretracefs.service:
```diff
@@ -0,0 +1,11 @@
+RemainAfterExit=yes
+
+[Install]
+WantedBy=multi-user.target
```
Newline

chaitu236 reviewed (changes requested):
![image](https://github.com/user-attachments/assets/42a117a9-a50a-4dee-9ab2-1a2a52a20277)

Instead of merging `nilrt/systemd/scarthgap` into your dev branch to get latest changes, we should rebase all the commits in your dev branch on top of the latest `nilrt/systemd/scarthgap`.

And then "mountdebugfs: Add service to services-nilrt" commit and it's revert can both be dropped.
